sql/src/main/java/io/crate/operation/collect/LuceneDocCollector.java

```diff
@@ -188,6 +198,14 @@ public void setNextReader(AtomicReaderContext context) throws IOException {
         }
     }
 
+    public void setNextOrderByValues(ScoreDoc scoreDoc) {
+        for (LuceneCollectorExpression expr : collectorExpressions) {
```

Comment:
if this method is called multiple times it might be worth creating a pre-filtered list which only contains OrderByCollectorExpressions